include/OGL/Repartitioner.H

```diff
@@ -0,0 +1,157 @@
+ * n ranks to m ranks with n>=m
+ */
+class Repartitioner {
+private:
+    using partition = gko::experimental::distributed::Partition<label, label>;
+
+    const label size_;  //! Size (n matrix rows) before repartitioning
+
+    const label repart_size_;  //! Size after repartitioning
+
+    const label ranks_per_gpu_;
+
+    const label verbose_;
+
+    std::shared_ptr<const partition> orig_partition_;
+
+public:
+    Repartitioner(label size, label ranks_per_gpu, label verbose,
+                  const ExecutorHandler &exec_handler)
+        : size_(size),
+          repart_size_(Repartitioner::compute_repart_size(size, ranks_per_gpu,
+                                                          exec_handler)),
+          ranks_per_gpu_(ranks_per_gpu),
+          verbose_(verbose),
+          orig_partition_(gko::share(
+              gko::experimental::distributed::build_partition_from_local_size<
+                  label, label>(exec_handler.get_ref_exec(),
+                                *exec_handler.get_communicator().get(),
+                                size))) {};
+
+    /* returns the owner rank for a given rank */
+    label get_owner_rank(label rank) const
+    {
+        return compute_owner_rank(rank, ranks_per_gpu_);
+    };
+
+    /* returns the owner rank for a given rank */
+    label get_owner_rank(const ExecutorHandler &exec_handler) const
+    {
+        return get_owner_rank(exec_handler.get_rank());
+    };
+
+    /* returns if current rank is an owner  */
+    bool is_owner(const ExecutorHandler &exec_handler) const
+    {
+        return exec_handler.get_rank() ==
+               get_owner_rank(exec_handler.get_rank());
+    };
+
+    /* @brief check if the given rank gets local after repartitioning
+     *
+     * */
+    bool reparts_to_local(const ExecutorHandler &exec_handler, label rank) const
+    {
+        return get_owner_rank(exec_handler) ==
+               compute_owner_rank(rank, ranks_per_gpu_);
+    };
+
+    label get_ranks_per_gpu() const { return ranks_per_gpu_; }
+
+    /* @brief computes the size after repartitioning
+     *
+     * Given a local size (can represent nrows or nnzs) the new size of
+     * this rank after repartitioning gets computed
+     */
+    static label compute_repart_size(label local_size, label ranks_per_gpu,
+                                     const ExecutorHandler &exec_handler);
+
+    label get_repart_size() const { return repart_size_; }
+
+    // TODO pass original dim as argument, if size can also be nnz
+    gko::dim<2> get_repart_dim() const
+    {
+        return gko::dim<2>{static_cast<gko::size_type>(repart_size_),
+                           static_cast<gko::size_type>(repart_size_)};
+    }
+
+    std::shared_ptr<const partition> get_orig_partition() const
+    {
+        return orig_partition_;
+    };
+
+
+    /* @brief given received interfaces this function sorts into local and non
+     *local
```

Comment:
It is a little confused here.

According to the comment here, the function sorts a given set of interfaces into local and nonlocal interfaces.
But the function takes local and nonlocal sparsity patterns and returns a new non-local sparsity pattern.

Does the function actually do a sorting job for a given set of interfaces?

Reply:
Good catch, this comment is outdated. I'll fix it with the PR that implements the function.